consensus/obcpbft/pbft-core.go

```diff
@@ -668,6 +671,17 @@ outer:
 		}
 		logger.Debug("Replica %d has detected request %s must be resubmitted", instance.id, d)
```

Comment:
Debug log without guard just produces garbage and spends cycles. I would suggest if logger.IsEnabledFor(logging.DEBUG) especially in high frequency paths.